pkg/errors/errors.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+// The idea of writing errors package in thanos is highly motivated from the Tast project of Chromium OS Authors. However, instead of
+// copying the package, we end up writing our own simplified logic borrowing some ideas from the errors and github.com/pkg/errors.
+// A big thanks to all of them.
+
+// Package errors provides basic utilities to manipulate errors with a useful stacktrace. It combines the
+// benefits of errors.New and fmt.Errorf world into a single package.
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// base is the fundamental struct that implements the error interface and the acts as the backbone of this errors package.
+type base struct {
+	info  string     // error message passed through calls like errors.Wrap, errors.Errorf
+	stack stacktrace // stacktrace where this error was generated
+	err   error      // actual error which
+}
+
+// Error implements the error interface.
+func (b *base) Error() string {
+	if b.err != nil {
+		return fmt.Sprintf("%s: %s", b.info, b.err.Error())
+	}
+	return b.info
+}
+
+// Unwrap implements the error Unwrap interface.
+func (b *base) Unwrap() error {
+	return b.err
+}
+
+// Format implements the fmt.Formatter interface to support the formatting of an error chain with "%+v" verb.
+func (b *base) Format(s fmt.State, verb rune) {
+	if verb == 'v' && s.Flag('+') {
+		s.Write([]byte(formatErrorChain(b)))
+		return
+	}
+
+	s.Write([]byte(b.Error()))
+}
+
+// New creates a new error with the given message and a stacktrace in details.
+// An alternative to errors.New function.
+func New(msg string) error {
+	return &base{
+		info:  msg,
+		stack: newStackTrace(),
+		err:   nil,
+	}
+}
+
+// Errorf creates a new error with the given message and a stacktrace in details.
+// An alternative to fmt.Errorf function.
+func Errorf(format string, args ...interface{}) error {
+	return &base{
+		info:  fmt.Sprintf(format, args...),
+		stack: newStackTrace(),
+		err:   nil,
+	}
+}
+
+// Wrap creates a new error with the given message, wrapping another error and a stacktrace in details.
+// If cause is nil, this is the same as New.
+func Wrap(cause error, msg string) error {
```

Comment:
Since we innovate here, I wonder if we have to have both `Wrapf` and `Wrap` at the end, why not having only `Wrapf` and just do `format` without `%v` or something? 🤔